src/include/storage/data_table.h

```diff
@@ -207,6 +208,11 @@ class DataTable {
    */
   DataTableCounter *GetDataTableCounter() { return &data_table_counter_; }
 
+  /**
+   * The tuple access strategy to access inside blocks in this datatable
+   */
+  const TupleAccessStrategy accessor_;
+
```

Comment:
Yeah, since the projected row we are passing in can have more than the number of columns the data table has. The check for number of columns in the projected row as to be done in sql_table. For this `TERRIER_ASSERT(out_buffer->NumColumns() <=
                     curr_dt_version.data_table->accessor_.GetBlockLayout().NumColumns() - NUM_RESERVED_COLUMNS,
                 "The output buffer never returns the version pointer columns, so it should have "
                 "fewer attributes.");` we need TupleAccessStrategy

Reply:
Modify this to just use the length of that column map for that table